TramsDataApi/Controllers/AcademyConversionProjectController.cs

```diff
@@ -1,46 +1,60 @@
 using TramsDataApi.UseCases;
 
 namespace TramsDataApi.Controllers
 {
     [ApiController]
-    public class AcademyConversionProjectController : ControllerBase
+	[Route("conversion-projects")]
+	public class AcademyConversionProjectController : ControllerBase
 	{
-        private readonly IUseCase<GetAcademyConversionProjectByIdRequest, AcademyConversionProjectResponse> _getAcademyConversionProjectById;
-        private readonly IUseCase<GetAllAcademyConversionProjectsRequest, IEnumerable<AcademyConversionProjectResponse>> _getAllAcademyConversionProjects;
+		private readonly IUseCase<GetAcademyConversionProjectByIdRequest, AcademyConversionProjectResponse> _getAcademyConversionProjectById;
+		private readonly IUseCase<GetAllAcademyConversionProjectsRequest, IEnumerable<AcademyConversionProjectResponse>> _getAllAcademyConversionProjects;
+		private readonly IUseCase<UpdateAcademyConversionProjectRequest, AcademyConversionProjectResponse> _updateAcademyConversionProject;
 
-        public AcademyConversionProjectController(
+		public AcademyConversionProjectController(
 			IUseCase<GetAcademyConversionProjectByIdRequest, AcademyConversionProjectResponse> getAcademyConversionProjectById,
-			IUseCase<GetAllAcademyConversionProjectsRequest, IEnumerable<AcademyConversionProjectResponse>> getAllAcademyConversionProjects)
-        {
-            _getAcademyConversionProjectById = getAcademyConversionProjectById;
-            _getAllAcademyConversionProjects = getAllAcademyConversionProjects;
-        }
-
-        [HttpGet]
-		[Route("conversion-projects")]
-        public IActionResult GetConversionProjects([FromQuery]int count = 50)
-        {
+			IUseCase<GetAllAcademyConversionProjectsRequest, IEnumerable<AcademyConversionProjectResponse>> getAllAcademyConversionProjects,
+			IUseCase<UpdateAcademyConversionProjectRequest, AcademyConversionProjectResponse> updateAcademyConversionProject)
+		{
+			_getAcademyConversionProjectById = getAcademyConversionProjectById;
+			_getAllAcademyConversionProjects = getAllAcademyConversionProjects;
+			_updateAcademyConversionProject = updateAcademyConversionProject;
+		}
+
+		[HttpGet]
+		public IActionResult GetConversionProjects([FromQuery] int count = 50)
+		{
 			// temporarily limiting count until we know rules around which to return as there's hundreds in db
 			var projects = _getAllAcademyConversionProjects.Execute(new GetAllAcademyConversionProjectsRequest { Count = count });
 
 			return Ok(projects);
 		}
 
-		[HttpGet]
-		[Route("conversion-projects/{id}")]
+		[HttpGet("{id}")]
 		public IActionResult GetConversionProjectById(int id)
 		{
 			var project = _getAcademyConversionProjectById.Execute(new GetAcademyConversionProjectByIdRequest { Id = id });
 			if (project == null)
-            {
+			{
 				return NotFound();
-            }
+			}
 
 			return Ok(project);
 		}
+
+		[HttpPatch("{id}")]
+		public IActionResult UpdateConversionProject(int id, UpdateAcademyConversionProjectRequest request)
```

Comment:
Should the id be coming from the request and not the model?

Reply:
it should indeed - should be sorted now